src/core/agent/services/signifyNotificationService.ts

```diff
@@ -102,6 +102,28 @@ class SignifyNotificationService extends AgentService {
         notifications.notes.shift();
       }
       for (const notif of notifications.notes) {
+        await this.markNotification(notif.i);
```

Comment:
This is marking it too early - we might crash or have an exception. We mark at the end when we are done processing or fully know we should skip this (e.g. for multi-sig)